src/runtime/Object.cpp

```diff
@@ -1943,6 +1943,18 @@ Value Object::getPrivateMember(ExecutionState& state, Object* contextObject, Ato
     return Value();
 }
 
+bool Object::hasPrivateMember(ExecutionState& state, Object* contextObject, AtomicString propertyName, bool shouldReferOuterClass)
+{
+    auto e = ensureExtendedExtraData();
+    auto piece = findPieceOnPrivateMemberChain(state, e, contextObject);
+    if (piece) {
+        return piece->m_privateMemberStructure->findProperty(propertyName);
+    } else if (shouldReferOuterClass && contextObject->asScriptClassConstructorFunctionObject()->outerClassConstructor()) {
+        return hasPrivateMember(state, contextObject->asScriptClassConstructorFunctionObject()->outerClassConstructor().value(), propertyName);
+    }
+    return false;
```

Comment:
There are two returns above this return, so I think this is dead code.

Reply:
I think `!piece && !(shouldReferOuterClass && contextObject->asScriptClassConstructorFunctionObject()->outerClassConstructor())` could be exists

Reply:
You are right. Sorry.

Reply:
I am fine :)

Reply:
I have a question about `shouldReferOuterClass` parameter.
Do we have to check outer class too?
And `shouldReferOuterClass` parameter is enabled by default, but actually called as false below

```c++
registerFile[code->m_dstIndex] = Value(right.toObject(state)->hasPrivateMember(state, state.findPrivateMemberContextObject(), AtomicString(state, left.asString()), false));
```

Reply:
Currently, there is no such case(shouldReferOuterClass is false)
but I added it for future && private member variable get operations could refer outer class